src/invocation/ClusterService.ts

```diff
@@ -131,15 +131,15 @@ export class ClusterService extends EventEmitter {
 
     private onConnectionClosed(connection: ClientConnection) {
         this.logger.warn('ClusterService', 'Connection closed to ' + connection.address.toString());
-        if (connection.address === this.getOwnerConnection().address) {
+        if (connection.address.equals(this.getOwnerConnection().address)) {
```

Comment:
can connection or address be null at here?

Reply:
No. `connectionClosed` event is emitted here: https://github.com/hazelcast/hazelcast-nodejs-client/blob/master/src/invocation/ClientConnectionManager.ts#L112 . It checks there exists such a connection before emitting the event.